Modules/_io/winconsoleio.c

```diff
@@ -556,14 +556,18 @@ read_console_w(HANDLE handle, DWORD maxlen, DWORD *readlen) {
     Py_BEGIN_ALLOW_THREADS
     DWORD off = 0;
     while (off < maxlen) {
-        DWORD n, len = min(maxlen - off, BUFSIZ);
+        DWORD n = (DWORD) - 1; 
+        DWORD len = min(maxlen - off, BUFSIZ);
         SetLastError(0);
         BOOL res = ReadConsoleW(handle, &buf[off], len, &n, NULL);
 
         if (!res) {
             err = GetLastError();
             break;
         }
+        if (n == (DWORD) - 1 && (err = GetLastError()) == ERROR_OPERATION_ABORTED) {
+            break;
```

Comment:
This looks good to me. @zooba, does this need a test? It's an uncommon edge case that I discovered randomly while experimenting with `CancelSynchronousIo`.

@ValeriyaSinevich, FYI `read_console_w` and `_PyOS_WindowsConsoleReadline` are still broken per [bpo-28166](https://bugs.python.org/issue28166) when `PyErr_CheckSignals` returns 0, i.e. if it's not the main thread or the SIGINT handler doesn't raise. It's an easy fix in the same functions if you want to work on it. Just  immediately `continue` to retry `ReadConsoleW`.

Reply:
I don't think it needs a test - it's a narrow enough fix that we're very unlikely to regress it.